scarb/src/ops/package.rs

```diff
@@ -156,6 +156,20 @@ fn list_one_impl(
 }
 
 fn prepare_archive_recipe(pkg: &Package) -> Result<ArchiveRecipe> {
+    ensure!(
+        pkg.manifest.targets.iter().any(|x| x.is_lib()),
+        formatdoc! {
+            r#"
+            package `{package_name}` does not provide a `[lib]` target.
+            if a package does not provide a library target, it cannot be used as a dependency.
+            help: add `[lib]` section to package manifest
+             --> Scarb.toml
+            +   [lib]
```

Comment:
- By convention, we do not use dots in error messages
- The `dependency` is confusing - we are only packaging local code
- We need to emphasize why we need to return an error - not defining `lib` is not an error per se, without the context. 
Maybe something like this? Not sure on exact wording.
```suggestion
            cannot archive package `{package_name}` without a `lib` target
            help: add `[lib]` section to package manifest
             --> Scarb.toml
            +   [lib]
```